app/pages/lab/workflows.jsx

```diff
@@ -7,116 +7,161 @@ import WorkflowCreateForm from './workflow-create-form';
 import LoadingIndicator from '../../components/loading-indicator';
 import Paginator from '../../talk/lib/paginator';
 
-const WorkflowsPage = (props) => {
-  const renderWorkflowTable = ((workflow) => {
+class WorkflowsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleWorkflowSettingChange = this.handleWorkflowSettingChange.bind(this);
+    this.state = {
+      active: false,
+      error: null
+    };
+  }
+
+  handleWorkflowSettingChange(workflow, e) {
+    const defaultWorkflow = (this.props.configuration) ? this.props.configuration.default_workflow : null;
+    const checked = e.target.checked;
+    workflow.update({ active: checked }).save()
+      .catch(error => console.log(error))
+      .then(() => {
+        if (!workflow.active && workflow.id === defaultWorkflow) {
+          this.props.project.update({ 'configuration.default_workflow': null });
+          this.props.project.save();
+        }
+      })
+      .then(() => this.setState({ active: workflow.active }));
+  }
+
+  renderWorkflowTable(workflow) {
```

Comment:
This render method is getting quite large. Perhaps it's time to break this out into its own functional component?

Reply:
Done.